### DISCUSSION TITLE
Different Monitor layout presets

### DISCUSSION THREAD

**Discussion by Yeraze** [nikitabobko/AeroSpace#765] [questions-and-answers]
I have 3 primary monitor layouts I use on my macbook:
* Built-in display only (when Traveling)
* Built-in display + 1 external monitor (Home Office)
* Built-in display + 2 external monitors (Work office)

Is there a way with Aerospace to have 3 defined configurations for these layouts, with application preset positions? (e.g. Which monitor and space contains Outlook, Kitty, Teams, Chrome, etc)

**Comment by alewis001** [comment]
I'm interested in this as well. Most of the time I have my macbook + 1 external monitor but occasionally I unplug the monitor, at which point the aerospace layout config is no longer appropriate. 

I have a 4k external monitor, I don't have apps maximised on that screen and I've set gaps for the "main" monitor (the 4k monitor is the "main" one via macos display settings when it's attached). Unfortunately, when the 4k monitor is disconnected the gaps are applied on the macbook screen as that becomes "main" which ideally wouldn't happen, instead I'd probably opt for all/most windows to be maximised.

**Comment by alewis001** [comment]
I think I've now just solved my issue, maybe this will help you @Yeraze 

I reworked my settings to...

```yaml
[gaps]
inner.horizontal = 5
inner.vertical =   5
outer.left =       [{monitor."PL3272UH" = 550}, 0]
outer.bottom =     [{monitor."PL3272UH" = 50}, 0]
outer.top =        [{monitor."PL3272UH" = 50}, 0]
outer.right =      [{monitor."PL3272UH" = 550}, 0]

[workspace-to-monitor-force-assignment]
Alt = ['built-in.*', 'main']
AltTesting = ['built-in.*', 'main']
OBS = ['built-in.*', 'main']
```

I used to set the gaps based on `montitor.main` but that meant that when the external monitor was unplugged and the laptop screen became "main", the gaps were wrong. Matching on the monitor (and now I've thought about it 🤦 ) makes more sense. Using the array of workspaces means that the workspaces in force assignment means that they will try to go to the built-in laptop screen first, otherwise will go to main. I technically don't need the array as "built-in" will be "main" if the external monitor is unplugged. I did it this way thinking that if I for some reason close the Laptop lid whilst the external monitor is still attached, I think the workspaces should appear on the external monitor properly (not actually tried it though).

HTH

**Comment by alewis001** [comment]
The comment in the default config might help, in case I'm misunderstanding what you want to do:

```
# Gaps between windows (inner-*) and between monitor edges (outer-*).
# Possible values:
# - Constant:     gaps.outer.top = 8
# - Per monitor:  gaps.outer.top = [{ monitor.main = 16 }, {monitor."some-pattern" = 32 }, 24]
#                 In this example, 24 is a default value when there is nomatch.
#                 Monitor pattern is the same as for 'workspace-to-monitor-force-assignment'.
#                 See: https://nikitabobko.github.io/AeroSpace/guide#assign-workspaces-to-monitors
```
I think it sounds like you want something like: `gaps.outer.top = [{ monitor."built-in.*" = 0 }, 40]`

As mentioned in the comment above, the last param in the array is the default, which I think in this case you want 40. You then set the override for the specific monitor. You could use `aerospace list-monitors` ("https://nikitabobko.github.io/AeroSpace/commands#list-monitors") to get their full names/ids as aerospace sees them instead of using a pattern in the `monitor.***` format. Obviously my example is just setting `gaps.outer.top`, you would need to set the other edges as desired.

HTH

**Reply by alewis001** [reply]
Not sure why your email didn't get posted here as a question 🤷 

Also, for some reason the formatting looks fine in preview but won't display when not editing 😔
